### ISSUE TITLE
Add Kong to supported network proxies on knative.dev

### ISSUE THREAD

**Issue by hbagdi** [Kong/kubernetes-ingress-controller#706]
Kong Ingress Controller has added support for Knative Ingress resource.
We should document this on Knative's documentation website as well.

**Comment by hbagdi** [comment]
WIP: https://github.com/knative/docs/pull/2341

**Comment by hbagdi** [comment]
Blocked on https://github.com/Kong/kubernetes-ingress-controller/issues/707

**Comment by kevinnesta** [comment]
Hi @hbagdi,
If you can describe in that documentation how to route traffic to a Knative service using specific path, I will be great! ^^

Currently I'm stuck on that, trying to query _https://mydomain/hello_ but getting 404.
The same request _https://mydomain/hello_ specifying the header _host: hello-service.default.mydomain.com_ is working fine!
Should not be so far from the truth but need some guidelines ^^

Thanks for your help

**Comment by hbagdi** [comment]
Hmm. Does knative support path based routing? I'm not aware of that being the case.

**Comment by kevinnesta** [comment]
I think you're right, but how can we specify the host header before reaching the upstream Knative service? I tried the Kong plugin 'request-transformer' but it looks we can't override 'host' header (looks to be ignored). 
My concern is then to have Knative integration fully functional for basic use case such as path based routing. Maybe you have a 'good way' to achieve that, with some recommendation that we can get from the documentation, what do you think?

Btw, thanks for the great job already done on that topic!

**Comment by hbagdi** [comment]
The original scope of this issue is now complete.

**Comment by MZohaibAsaad** [comment]
@kevinnesta  Can you please write a blog or share your yamls here so that i can follow it. i'm stuck at the exact same scenario for past 4 weeks. i'm new to k8s domain, so don't have enough knowledge to solve it myself yet. I understand what should be done here, but don't know how to do it. i did try creating an Ingress and redirecting to kong-proxy service but it failed for me. It will be a great favor if you could guide me. Thanks in advance.

**Comment by kevinnesta** [comment]
Hi @MZohaibAsaad ,

I never succeeded this, only way remove Knative from my service... Because I was trying to get Ingress working with OIDC Plugins, I faced infinite loops. Anyway, here are my try without OIDC, hope it'll help you

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: hello-ingress
  namespace: kong
  annotations:
    kubernetes.io/ingress.class: "kong"
    konghq.com/plugins: rewrite-host-header-hello
spec:
  rules:
    - host:
      http:
        paths:
          - path: /hello
            backend:
              serviceName: kong-proxy
              servicePort: 80
---
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: rewrite-host-header-hello
  namespace: kong
plugin: request-transformer
config:
  replace:
    headers:
      - 'host:hello.default.<DOMAIN>'
---
apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  name: hello
  namespace: default
spec:
  template:
    metadata:
      annotations:
        ingress.kubernetes.io/service-upstream: "true"
        autoscaling.knative.dev/minScale: "1"
    spec:
      containers:
        - image: gcr.io/knative-samples/helloworld-go
          env:
            - name: TARGET
              value: Go Sample v1
```

As you can see, the trick is to re-route the traffic on path '_/hello_' to the _**kong-proxy**_ service rewriting the header to match Knative expectation.
FYI, <DOMAIN> tag should be replaced by your Knative declared domain.
Let me know if it's working for your UC

**Comment by MZohaibAsaad** [comment]
Thanks @kevinnesta. I'm trying it out right now. i was doing pretty much same, except the replace header part, i was adding it. i'll tell you of the result.

**Comment by MZohaibAsaad** [comment]
Thank you so much @kevinnesta. The above yaml worked for me with minor changes. The key part which i was missing was deploying the ingress and plugin in "kong" namespace. i was deploying them in default namespace. Here is the yaml for someone else who stumbles on this issue: (Used a sample echo server)
```

apiVersion: serving.knative.dev/v1
kind: Service
metadata:
  name: knative-service-test
  namespace: default
spec:
  template:
    spec:
      containers:
        - image: ealen/echo-server:latest
---
apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: request-transformer-example
  namespace: kong
plugin: request-transformer
config:
  replace:
    headers:
      - 'host:knative-service-test.default.example.com'
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: demo
  namespace: kong
  annotations:
    kubernetes.io/ingress.class: kong
    plugins.konghq.com: request-transformer-example
spec:
  rules:
  - http:
      paths:
      - path: /temp
        backend:
          serviceName: kong-proxy
          servicePort: 80
```

**Comment by mannro** [comment]
@kevinnesta what if i need prevent host in calling service e.g. in matter of mutlitenancy?
